packages/babel-plugin/src/utils/build-styled-component.ts

```diff
@@ -102,6 +101,60 @@ const traverseStyledBinaryExpression = (node: t.BinaryExpression, nestedVisitor:
   return node;
 };
 
+/**
+ * Returns a list of destructured props used in expressions of a styled component
+ *
+ * For example:
+ * ```
+ * const Component = styled.div`
+ *  width: $({ width }) => `${width}px`;
+ *  height: $({ height }) => `${height}px`;
+ * `;
+ * ```
+ * Returns list of [width, height]
+ *
+ * @param node {t.Node} Node of the styled component
+ */
+const getDestructuredProps = (node: t.Node): string[] => {
+  const destructuredProps: string[] = [];
+
+  const appendDestructuredPropsVisitor = {
+    noScope: true,
+    ObjectPattern(path: NodePath<t.ObjectPattern>) {
+      const { listKey, key } = path;
+
+      if (listKey === 'params' && key === 0) {
+        // Uses the destructure ObjectPattern path to get all the prop references, i.e.
+        // { width } becomes width
+        // { width, height } becomes width,height
+        // { size: { width } } becomes size:{width}
+        // { width: alias } becomes width:alias
+        const propsUsed = path.toString().replace(/\s/g, '').slice(1, -1).split(',');
+
+        destructuredProps.push(...propsUsed);
+        path.stop();
+      }
+    },
+  };
+
+  const arrowFunctionVisitor = {
```

Comment:
I would inline this in the below `traverse`

Reply:
👍  Ah I see it's only beneficial to split out nested visitors.